workflows/postprocessing.nf

```diff
@@ -174,17 +128,27 @@ workflow POSTPROCESSING {
 
     writemeta()
     filtered = EXCLUDE_MNPS(ch_samplesheet).ch_output_excludemnps
-                    .map{meta, files -> tuple( groupKey(meta.familyId, meta.sampleSize),meta,files)}
-                    .groupTuple()
-                    .map{ familyId, meta, files -> //now that samples are grouped together, we no longer follow sample in meta
-                        [
-                            meta[0].findAll{it.key != "sample"}, //meta
-                            files.flatten()]}                     //files
-    //Using 2 as threshold because we have 2 files per patient (gcvf.gz, gvcf.gz.tbi)
-    filtered_one = filtered.filter{it[0].sampleSize == 1}
+        .map{meta, files -> tuple(groupKey(meta.familyId, meta.sampleSize),meta,files[0],files[1])}
+        .groupTuple()
+        .map{ familyId, meta, vcf, tbi -> 
+        //now that samples are grouped together, we no longer follow sample in meta, and the id no longer needs the sampleId
+            [
+                meta[0].findAll{it.key != "sample"}.findAll{it.key != "id"}, //meta
+                vcf.flatten(),
+                tbi.flatten(),                                               //files
+            ]}
+        .map{meta,vcf,tbi -> 
+            [
+                meta+[id: meta.familyId],vcf,tbi               //we replace the sampleID that was removed
+            ]}
```

Comment:
I think it would simplify the code to pre-create the new meta in a local variable:
```
.map {familyId, meta, vcf, tbi ->
    def updated_meta = meta[0].findAll{!["sample", "id"].contains(it.key) }
    updated_meta["id"] = updated_meta.familyId
    [ updated_meta, vcf.flatten(), tbi.flatten()]
```

Reply:
nice one